iroh/src/endpoint.rs

```diff
@@ -1510,6 +1510,11 @@ impl Connecting {
             Err(_) => bail!("unknown handshake type"),
         }
     }
+
+    /// The remote_node_id, this should always be available after a successfull call to ['Connecting::handshake_data'] or ['Connecting::alpn']
```

Comment:
Is this really true?  Looking at the code I think this is populated when the `Connecting` is created via `Endpoint::connect` but not at all when created via `Endpoint::accept`.

Looking at `Connection::remote_node_id` it seems you need the peer identity data to be able to get the remote node id.  I think you have to dig deeper into rustls to get the nodeid out at this stage.  It would be very cool if you could figure out how to do that though!

I think you should also write a unittest for this.